src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -14,15 +14,24 @@ type private Format = ReadOnlyMemory<byte>
 [<Struct>]
 type ExpectedVersion = Any | StreamVersion of int64
 
+module private Queries =
+    [<Literal>]
+    let writeMessage = "select * from write_message($1::text, $2, $3, $4, $5, $6)"
+    [<Literal>]
+    let readLast = "select position, type, data, metadata, id::uuid, time from get_last_stream_message($1, $2);"
+    [<Literal>]
+    let readStream = "select position, type, data, metadata, id::uuid, time from get_stream_messages($1, $2, $3);"
+
+
 module private Sql =
-    let addExpectedVersion name (value: ExpectedVersion) (p: NpgsqlParameterCollection) =
+    let addExpectedVersion (value: ExpectedVersion) (p: NpgsqlParameterCollection) =
         match value with
-        | StreamVersion value -> p.AddWithValue(name, NpgsqlDbType.Bigint, value) |> ignore
-        | Any                 -> p.AddWithValue(name, NpgsqlDbType.Bigint, DBNull.Value) |> ignore
-    let addNullableString name (value: string option) (p: NpgsqlParameterCollection) =
+        | StreamVersion value -> p.AddWithValue(NpgsqlDbType.Bigint, value) |> ignore
+        | Any                 -> p.AddWithValue(NpgsqlDbType.Bigint, DBNull.Value) |> ignore
+    let addNullableString (value: string option) (p: NpgsqlParameterCollection) =
         match value with
-        | Some value -> p.AddWithValue(name, NpgsqlDbType.Text, value) |> ignore
-        | None       -> p.AddWithValue(name, NpgsqlDbType.Text, DBNull.Value) |> ignore
+        | Some value -> p.AddWithValue(NpgsqlDbType.Text, value) |> ignore
```

Comment:
could just `defaultArg` ?

Reply:
Or Maybe convert this to a type augmentation (with autoopen as the module is provate anyway)
type ParamCol with
    member p.AddNullableString(t, v) = p.AddWithValue(t, defaultArg v DBNull.value) |> ignore

Reply:
> could just defaultArg?

Not quite. DbNull is not a string unfortunately so `defaultArg strOpt DbNull` won't compile

> Or Maybe convert this to a type augmentation (with autoopen as the module is provate anyway)

Done